document-store/src/main/java/org/hypertrace/core/documentstore/postgres/utils/PostgresUtils.java

```diff
@@ -215,6 +215,76 @@ public static String parseNonCompositeFilter(
     return filters;
   }
 
+  public static String prepareParsedNonCompositeFilter(
```

Comment:
Note: we can combine this method with the existing `parseNonCompositeFilter`. However, I had to add a few if-clauses. So, I will do the clean (or refactor) as follow-up.

Reply:
Can we add a `// TODO:` comment so we can keep track?

Reply:
Done.